src/compat.ml

```diff
@@ -0,0 +1,40 @@
+module Option = struct
+  type 'a t = 'a option = None | Some of 'a
+
+  let is_some = function None -> false | Some _ -> true
+
+  let value_exn = function
+    | None -> failwith "Option.value_exn None"
+    | Some x -> x
+
+  let join_list l =
+    if List.for_all Option.is_some l then Some (List.map value_exn l) else None
+end
+
+module Char = struct
+  include Char
+
+  let equal x y = compare x y = 0
```

Comment:
This is the slow polymorphic version of the function. I'd prefer if this wasn't defined. The better version is:
```ocaml
let equal (x : char) y = x = y
```